pycyphal/dsdl/_compiler.py

```diff
@@ -171,14 +170,28 @@ def compile(  # pylint: disable=redefined-builtin
         (root_namespace_name,) = set(map(lambda x: x.root_namespace, composite_types))  # type: ignore
         _logger.info("Read %d definitions from root namespace %r", len(composite_types), root_namespace_name)
 
-        # Generate code
-        assert isinstance(output_directory, pathlib.Path)
         root_ns = nunavut.build_namespace_tree(
             types=composite_types,
             root_namespace_dir=str(root_namespace_directory),
             output_dir=str(output_directory),
             language_context=language_context,
         )
+    else:
+        root_ns = nunavut.build_namespace_tree(
+            types=[],
+            root_namespace_dir=str(""),
+            output_dir=str(output_directory),
+            language_context=language_context,
+        )
+
+    lockfile = Locker(
+        root_namespace_name=root_namespace_name if root_namespace_name else "support",
+        output_directory=output_directory,
+    )
+    lockfile_created = lockfile.create()
+    if lockfile_created:
```

Comment:
The next step is to introduce a context manager. Currently, while this code is logically correct, it is not exception-safe.

```suggestion
    if lockfile.create():
```